rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueWriter.scala

```diff
@@ -0,0 +1,271 @@
+/*
+*************************************************************************************
+* Copyright 2017 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+*
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+
+package com.normation.rudder.ncf
+
+import net.liftweb.common.Loggable
+import java.nio.file.Paths
+import java.nio.file.Files
+import scala.xml.NodeSeq
+import java.nio.charset.StandardCharsets
+import com.normation.inventory.domain.AgentType
+import net.liftweb.common.Box
+import com.normation.rudder.repository.xml.GitArchiverUtils
+import com.normation.cfclerk.services.GitRepositoryProvider
+import java.io.File
+import com.normation.rudder.repository.xml.RudderPrettyPrinter
+import com.normation.rudder.repository.GitModificationRepository
+import com.normation.rudder.repository.xml.GitArchiverFullCommitUtils
+import com.normation.eventlog.ModificationId
+import org.eclipse.jgit.lib.PersonIdent
+import com.normation.rudder.repository.GitPath
+import net.liftweb.common.Full
+import com.normation.eventlog.EventActor
+import com.normation.rudder.services.user.PersonIdentService
+
+class TechniqueWriter(
+  archiver : TechniqueArchiver
+) extends Loggable {
+
+  private[this] var agentSpecific = new ClassicTechniqueWriter :: new DSCTechniqueWriter :: Nil
+
+  def commonMetadata(technique : Technique, methods: Map[BundleName, GenericMethod]) = {
```

Comment:
please add types on public sig. It helps both the reader and the compiler speed.

Reply:
(and it's interesting to know it's an XMLNode, not a nodeseq, because transformation on simple node are simpler)

Reply:
Yes, totally ok with this, jsut i did not know the type i would return so wanted to add it at the end ...

I think i should encapsulate this in an error type